mobilecoind/src/service.rs

```diff
@@ -599,9 +599,16 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
                 )
             })?;
 
-        let tx_out = self.ledger_db.get_tx_out_by_index(index).map_err(|err| {
-            rpc_internal_error("ledger_db.get_tx_out_by_index", err, &self.logger)
-        })?;
+        let tx_out = self
+            .ledger_db
+            .get_tx_out_by_index(index)
+            .map_err(|err| match err {
+                LedgerError::NotFound => RpcStatus::with_message(
+                    RpcStatusCode::NOT_FOUND,
+                    format!("tx_out {index} not found"),
+                ),
```

Comment:
Good catch, I missed that one. Seems like this one should in fact be an internal error since something's weird with the ledger if the first one succeeded and the second one failed.